.github/actions/clp-core-build-containers/action.yaml

```diff
@@ -0,0 +1,91 @@
+name: "clp-core-build-containers"
+description: "Builds and publishes container images for CLP-core's dependencies and binaries"
+
+inputs:
+  os_name:
+    description: "Name of container OS"
+    required: true
+  clp_core_dir:
+    description: "CLP-core's component directory"
+    required: true
+  push_deps_image:
+    description: "Whether to publish a container image containing CLP's dependencies"
+    required: true
+  push_binaries_image:
+    description: "Whether to publish a container image containing CLP's binaries"
+    required: true
+  local_registry_port:
+    description: "Port number for the local registry"
+    required: true
+  token:
+    description: "Registry token"
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - uses: "docker/setup-buildx-action@v3"
+      with:
+        driver-opts: "network=host"
+
+    - if: "inputs.push_deps_image == 'true'"
+      uses: "docker/login-action@v3"
+      with:
+        registry: "ghcr.io"
+        username: "${{github.actor}}"
+        password: "${{inputs.token}}"
+
+    - id: "get_deps_image_props"
+      shell: "bash"
+      run: |
+        if [[ "${{inputs.push_deps_image}}" == "true" ]] ; then
+          registry=ghcr.io
+          branch=${{github.ref_name}}
+        else
+          registry=localhost:${{inputs.local_registry_port}}
+          branch=latest
+        fi
+
+        echo "REGISTRY=$registry" >> "$GITHUB_OUTPUT"
+        echo "BRANCH=$branch" >> "$GITHUB_OUTPUT"
+
+    - id: "deps_image_meta"
+      uses: "docker/metadata-action@v5"
+      with:
+        images: "${{steps.get_deps_image_props.outputs.REGISTRY}}\
+          /${{github.repository}}\
+          /clp-core-dependencies-x86-${{inputs.os_name}}"
+        tags: |
+          type=raw,value=${{steps.get_deps_image_props.outputs.BRANCH}}
+
+    - uses: "docker/build-push-action@v5"
+      with:
+        context: "${{inputs.clp_core_dir}}"
+        file: "${{inputs.clp_core_dir}}/tools/docker-images/clp-env-base-${{inputs.os_name}}\
+          /Dockerfile"
+        push: true
+        tags: "${{steps.deps_image_meta.outputs.tags}}"
+        labels: "${{steps.deps_image_meta.outputs.labels}}"
+
+    - uses: "./.github/actions/clp-core-build"
+      with:
+        image_name: "${{steps.deps_image_meta.outputs.tags}}"
+
+    - if: "inputs.push_binaries_image == 'true'"
```

Comment:
I'm not sure if you can put constants first in GitHub Actions, or if it even matters as you can't accidentally assign a value in a GitHub Action or Bash. But, just putting this here if you want consistency with c++ style.

Reply:
Yeah, GitHub doesn't allow constants first for whatever reason. I think they allow it inside `${{}}` but to avoid confusion, I just keep it consistent as `var op constant`.